app/components/channel_drawer/channel_list.js

```diff
@@ -336,8 +463,27 @@ class ChannelList extends React.Component {
             );
         }
 
+        const modalOptions = (
+            <ModalOptions
+                title={this.state.modalTitle}
+                options={this.state.options}
+                visible={this.state.showOptions}
+                onChange={(option) => {
```

Comment:
Oh wow, bad comment now that I look at it. I meant to add that it didn't stop re-rendering until I extended PureComponent and moved the function that is in the onChange property to a method on channel_list.